src/plugins/vis_type_table/server/usage_collector/get_stats.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * and the Server Side Public License, v 1; you may not use this file except in
+ * compliance with, at your election, the Elastic License or the Server Side
+ * Public License, v 1.
+ */
+
+import { get } from 'lodash';
+import { ElasticsearchClient } from 'kibana/server';
+import { SearchResponse } from 'elasticsearch';
+import { SavedVisState } from 'src/plugins/visualizations/common';
+import { TableVisParams, VIS_TYPE_TABLE } from '../../common';
+
+type ESResponse = SearchResponse<{ visualization: { visState: string } }>;
+
+export interface VisTypeTableUsage {
+  /**
+   * Total number of table type visualizations
+   */
+  total: number;
+  /**
+   * Total number of table visualizations, using "Split table" agg
+   */
+  total_split: number;
+  /**
+   * Split table by columns stats
+   */
+  split_columns: {
+    total: number;
+    enabled: number;
+  };
+  /**
+   * Split table by rows stats
+   */
+  split_rows: {
+    total: number;
+    enabled: number;
+  };
+}
+
+/*
+ * Parse the response data into telemetry payload
+ */
+export async function getStats(
+  esClient: ElasticsearchClient,
+  index: string
+): Promise<VisTypeTableUsage | undefined> {
+  const searchParams = {
+    size: 10000, // elasticsearch index.max_result_window default value
+    index,
+    ignoreUnavailable: true,
+    filterPath: ['hits.hits._source.visualization'],
+    body: {
+      query: {
+        bool: { filter: { term: { type: 'visualization' } } },
+      },
+    },
+  };
+  const { body: esResponse } = await esClient.search<ESResponse>(searchParams);
```

Comment:
NIT: Shouldn't we use the `soClient` (also provided in the `fetch` context instead of searching to the `.kibana` index? AFAIK, ES is planning to forbid direct access to the system indices in the future, and I don't think we are doing anything that the SO.find API doesn't support (i.e.: other plugins run aggregations).

Reply:
hey @afharo , thanks for the hint!! 🙏 
Updated with the `soClient`, code looks much more better now!